vite.config.js

```diff
@@ -0,0 +1,21 @@
+const path = require("path");
```

Comment:
I didn't find what browsers/syntax we are targeting with the build. Is any modern syntax transpiled?

We are using some "modern" features such as let/const, object shorthand, etc.. which are probably all well supported by now, but wanted to understand if the sources are transpiled in any way and which syntax is safe to use.

Reply:
It's transpiling modern syntax, yes. E.g:

```javascript
options?.onWidgetLoad ?? false
```

becomes:
```javascript
var _a;
_a = options == null ? void 0 : options.onWidgetLoad) != null ? _a : false
```

You can read about the build target here:
https://vitejs.dev/guide/build.html#browser-compatibility

Reply:
Slightly worried that it might break in ancient browsers, but I guess we are using the same build config for the JS API and have already deployed a version with that distribution?

Ideally, I would like to base this off [ftrack-browserslist-config/](https://bitbucket.org/ftrack/ftrack-browserslist-config/), but vite does not support that: https://github.com/vitejs/vite/discussions/6849

Reply:
We might be able to use something like https://github.com/nihalgonsalves/esbuild-plugin-browserslist (taken from https://github.com/evanw/esbuild/issues/121), but i'm a bit worried of the correctness and low usage of the plugin. I think I would prefer if we could use the vite defaults. 

Yes, this is the same we use for javascript-api.

If we have a use case where we need to support ancient browsers, I suggest we leave it to the user of this library to transpile the application to a compatible format (as will be the case in our frontend).

Reply:
I assume we will encounter issues in the JS API before anything using this library and the defaults are fairly conservative, so seems like a safe change.

This is mainly a theoretical concern for web views / embedded browsers in desktop applications. The [VFX reference platform](https://vfxplatform.com/) has Qt 5.15.x as a guideline, which has Chromium 87 (the minimum version by vite) as staring with 5.15.3. https://wiki.qt.io/QtWebEngine/ChromiumVersions